src/index.ts

```diff
@@ -39,6 +40,81 @@ const {grpc} = new gax.GrpcClient(opts);
  */
 const PROJECT_ID_PLACEHOLDER = '{{projectId}}';
 
+// tslint:disable-next-line:no-any
+export type Metadata = any;
+
+/**
+ * @typedef {array} CreateTopicResponse
+ * @property {Topic} 0 The new {@link Topic}.
+ * @property {object} 1 The full API response.
+ */
+export type CreateTopicResponse = [Topic, object];
+
+/**
+ * @callback CreateTopicCallback
+ * @param {?Error} err Request error, if any.
+ * @param {Topic} topic The new {@link Topic}.
+ * @param {object} apiResponse The full API response.
+ */
+export interface CreateTopicCallback {
+  (err?: Error|null, topic?: Topic|null, apiResponse?: object): void;
+}
+
+export type Client = 'PublisherClient'|'SubscriberClient';
+
+export interface RequestConfig {
+  client: Client;
+  method: string;
+  reqOpts?: object;
```

Comment:
Aren't these `DecorateRequestOptions`?

Reply:
Nope - this is a rather custom object unique to the PubSub library as a hybrid (gapic + ts-hand-written layer).

`client` is the GAPIC client name (as a`string`), e.g. PublisherClient, exposed in `src/v1/publisher_client.js`
`method` is the method name (as a `string`) of the client to call - e.g. `method: listTopics` would invoke `PublisherClient.listTopics`
`reqOpts` are the options to the rpc method, [example](https://cloud.google.com/pubsub/docs/reference/rpc/google.pubsub.v1#google.pubsub.v1.GetTopicRequest), which depends on which method it is.
`gaxOpts` are configuration to gax and gRPC (autoPaginate, retry etc..).